src/main/java/bio/terra/service/cohortbuilder/HintsService.java

```diff
@@ -1,151 +1,124 @@
-                            new DisplayHintEnum()
-                                .enumHintValues(
-                                    Arrays.asList(
-                                        new DisplayHintEnumEnumHintValues()
-                                            .enumVal(
-                                                new ValueDisplay()
-                                                    .display("Hispanic or Latino")
-                                                    .value(
-                                                        new Literal()
-                                                            .dataType(DataType.INT64)
-                                                            .valueUnion(
-                                                                new LiteralValueUnion()
-                                                                    .int64Val(38003563L))))
-                                            .count(54453),
-                                        new DisplayHintEnumEnumHintValues()
-                                            .enumVal(
-                                                new ValueDisplay()
-                                                    .display("Not Hispanic or Latino")
-                                                    .value(
-                                                        new Literal()
-                                                            .dataType(DataType.INT64)
-                                                            .valueUnion(
-                                                                new LiteralValueUnion()
-                                                                    .int64Val(38003564L))))
-                                            .count(2272403))))))
-        .addDisplayHintsItem(
-            new DisplayHint()
-                .attribute(
-                    new Attribute()
-                        .name("year_of_birth")
-                        .type(Attribute.TypeEnum.SIMPLE)
-                        .dataType(DataType.INT64))
-                .displayHint(
-                    new DisplayHintDisplayHint()
-                        .numericRangeHint(
-                            new DisplayHintNumericRange().min(1909.0D).max(1983.0D))));
+        .sql(sql)
+        .displayHints(
+            displayHints.entrySet().stream()
+                .map(
+                    attrHint -> {
+                      Attribute attr = entity.getAttribute(attrHint.getKey());
+                      DisplayHint hint = attrHint.getValue();
+                      return new bio.terra.model.DisplayHint()
+                          .attribute(ToApiConversionUtils.toApiObject(attr))
+                          .displayHint(hint == null ? null : toApiObject(hint));
+                    })
+                .collect(Collectors.toList()));
+  }
+
+  private DisplayHintDisplayHint toApiObject(DisplayHint displayHint) {
```

Comment:
Out of curiosity, was this a tanagra naming convention?

Reply:
Yeah, I think it was originally `ApiDisplayHintV2DisplayHint`, but I pulled the Api and V2